packages/pluggableWidgets/tree-view-web/src/components/TreeView.tsx

```diff
@@ -167,25 +188,43 @@ function TreeViewBranch(props: TreeViewBranchProps): ReactElement {
         }
     }, []);
 
-    const toggleTreeViewContent = useCallback(() => {
-        if (!isActualLeafNode) {
-            setTreeViewState(treeViewState => {
-                if (treeViewState === TreeViewState.LOADING) {
-                    // TODO:
-                    return treeViewState;
-                }
-                if (treeViewState === TreeViewState.COLLAPSED_WITH_JS) {
-                    return props.shouldLazyLoad ? TreeViewState.LOADING : TreeViewState.EXPANDED;
-                }
-                if (treeViewState === TreeViewState.COLLAPSED_WITH_CSS) {
-                    return TreeViewState.EXPANDED;
-                }
-                return props.shouldLazyLoad ? TreeViewState.COLLAPSED_WITH_CSS : TreeViewState.COLLAPSED_WITH_JS;
-            });
-        }
-    }, [isActualLeafNode, props.shouldLazyLoad]);
+    const treeViewBranchRef = useRef<HTMLLIElement>(null);
+    const { changeFocus } = props;
+
+    const eventTargetIsNotCurrentBranch = useCallback<(event: SyntheticEvent<HTMLElement>) => boolean>(event => {
+        const target = event.target as Node;
+        return (
+            !treeViewBranchRef.current?.isSameNode(target) &&
+            !treeViewBranchRef.current?.firstElementChild?.contains(target) &&
+            !treeViewBranchRef.current?.lastElementChild?.isSameNode(target)
+        );
+    }, []);
+
+    const toggleTreeViewContent = useCallback<ReactEventHandler<HTMLLIElement>>(
+        event => {
+            if (eventTargetIsNotCurrentBranch(event)) {
+                return;
+            }
+            if (!isActualLeafNode) {
+                setTreeViewState(treeViewState => {
+                    if (treeViewState === TreeViewState.LOADING) {
+                        // TODO:
```

Comment:
🤔

Reply:
not entirely sure whether we want some alternative behaviour when clicking on a tree node that is loading. I assume not, but for now I added a TODO just in case

Reply:
👍🏻